code/mob/living/critter/spider.dm

```diff
@@ -495,7 +495,10 @@
 			return
 		var/defenders = 0		//this is the amount of babies that will defend you
 		var/count = 0
-		for (var/mob/living/critter/spider/clown/CS in babies)
+		for (var/datum/weakref/ref as anything in babies)
+			var/mob/living/critter/spider/clown/CS = ref.deref()
+			if (CS == null)
+				babies.Remove(ref)
```

Comment:
you'll want to `continue` here or it's probably still going to runtime below (might get caught by `GET_DIST`, but I wouldn't rely on it)